lib/presentation/atoms/timer_atom.dart

```diff
@@ -18,25 +20,60 @@ class TimerAtom extends StatelessWidget {
     final ThemeData themeData = Theme.of(context);
     final ColorScheme colorScheme = themeData.colorScheme;
 
-    return CircularCountDownTimer(
-      duration: timer.inSeconds,
-      controller: controller,
-      width: MediaQuery.of(context).size.width / 2,
-      height: MediaQuery.of(context).size.height / 2,
-      ringColor: Colors.grey[300]!,
-      fillColor: Colors.purpleAccent[100]!,
-      strokeWidth: 20.0,
-      strokeCap: StrokeCap.round,
-      textStyle: TextStyle(
-        fontSize: 33.0,
-        color: colorScheme.onBackground,
-        fontWeight: FontWeight.bold,
-      ),
-      textFormat: CountdownTextFormat.MM_SS,
-      isReverseAnimation: true,
-      isReverse: true,
-      onComplete: onComplete,
-
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 40.0),
+      child: CustomTimer(
+          controller: controller,
+          builder: (state, time) {
+            return SizedBox(
+              width: min(MediaQuery.of(context).size.width / 2, MediaQuery.of(context).size.height / 2),
+              child: AspectRatio(
+                aspectRatio: 1.0,
+                child: ValueListenableBuilder(
+                  valueListenable: controller.state,
+                  builder: (context, value, child) {
+                    return GestureDetector(
+                      onTap: () {
+                        if (value == CustomTimerState.counting) {
+                          controller.pause();
+                        } else {
+                          controller.start();
+                        }
+                      },
+                      child: Stack(
+                        children: [
+                          Positioned.fill(
+                            child: CircularProgressIndicator(
+                              backgroundColor: Colors.grey[300],
+                              color: value == CustomTimerState.counting
+                                  ? Colors.purpleAccent[100]!
+                                  : Colors.grey[500],
```

Comment:
Colors should arive from theme

Reply:
https://github.com/guyluz11/infinite_horizons/assets/39937989/a3a403db-e0a2-47cc-b4ab-b6fe0101ae48

```
 backgroundColor: colorScheme.primaryContainer,
      color: value == CustomTimerState.counting
            ? colorScheme.primary
            : colorScheme.outline,
```
how about these colours?

Reply:
Good, thanks